### ISSUE TITLE
React - Infinite-scroll loading entities twice after entity update

### ISSUE THREAD

**Issue by ecostanzi** [jhipster/generator-jhipster#7927]
##### **Overview of the issue**

If Infinite scroll is enabled on an entity, after I update one of them, the FE loads the list twice after an entity update. The first part of the list contains the old entities, the second part contains the first page, containing the UPDATED entity.

##### **Reproduce the error**

I detected the problem in a production like environment (war file build with ./mvnw -Pprod package)

- Load enough entitities to trigger the infinite scroll. 
- Open the entity page and scroll to bottom (trigger the infinite scroll)
- click on edit on the first entity, edit the entity, click on save

The browser opens the entity list page, you can see:
- that the first line hasn't changed
- that the entity list is loaded twice, the second group in the list contains the updated entity

here's the [video](https://drive.google.com/open?id=168CSD8j4hfcvnrjeaisa6P6rrl7-omFM)

##### **Suggest a Fix**
After the HTTP PUT that updates the entitites I can see three subsequent calls to:
- http://localhost:8086/api/employees?page=0&size=20&sort=id,asc&cacheBuster=1530701104820 (returns the old entity)
- http://localhost:8086/api/employees?page=1&size=20&sort=id,asc&cacheBuster=1530701104839 (returns the second page)
- http://localhost:8086/api/employees?cacheBuster=1530701104875 (returns the first page, with the *updated* entities)

##### **JHipster Version(s)**

5.0.1

##### **JHipster Version(s)**

```
humanresources@0.0.0 C:\Users\ecostanzi\dev\learn\jhipster\jhipster5-react-monolith
`-- (empty)

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "it.intesys.lab",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "5.0.1",
    "applicationType": "monolith",
    "baseName": "humanresources",
    "packageName": "it.intesys.lab",
    "packageFolder": "it/intesys/lab",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "mariadb",
    "prodDatabaseType": "mariadb",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "replaced-by-jhipster-info",
    "clientFramework": "react",
    "useSass": true,
    "clientPackageManager": "yarn",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "it"
    ]
  }
}
</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Region (region) {
  regionName String
}
entity Country (country) {
  countryName String
}
/**
 * not an ignored comment
 */
entity Location (location) {
  streetAddress String,
  postalCode String,
  city String,
  stateProvince String
}
entity Department (department) {
  departmentName String required
}
/**
 * Task entity.
 * @author The JHipster team.
 */
entity Task (task) {
  title String,
  description String
}
/**
 * The Employee entity.
 */
entity Employee (employee) {
  /**
   * The firstname attribute.
   */
  firstName String,
  lastName String,
  email String,
  phoneNumber String,
  hireDate Instant,
  salary Long,
  commissionPct Long
}
entity Job (job) {
  jobTitle String,
  minSalary Long,
  maxSalary Long
}
entity JobHistory (job_history) {
  startDate Instant,
  endDate Instant,
  language Language
}

enum Language {
  FRENCH,
  ENGLISH,
  SPANISH
}

relationship OneToOne {
  Country{region(regionName)} to Region,
  Location{country(countryName)} to Country,
  Department{location(city)} to Location,
  JobHistory{job(jobTitle)} to Job,
  JobHistory{department(departmentName)} to Department,
  JobHistory{employee(email)} to Employee
}
relationship OneToMany {
  /**
   * A relationship
   */
  Department{employee} to Employee{department(departmentName)},
  Employee{job} to Job{employee(email)}
}
relationship ManyToOne {
  Employee{manager(email)} to Employee
}
relationship ManyToMany {
  Job{task(title)} to Task{job}
}

dto Region, Country, Location, Department, Task, Employee, Job, JobHistory with mapstruct
service Region, Country, Location, Department, Task, Employee, Job, JobHistory with serviceImpl
paginate Employee, JobHistory with infinite-scroll
paginate Job with pagination

</pre>
</details>

##### **Environment and Tools**

java version "9.0.4"
Java(TM) SE Runtime Environment (build 9.0.4+11)
Java HotSpot(TM) 64-Bit Server VM (build 9.0.4+11, mixed mode)

git version 2.12.2.windows.2

node: v8.11.3

npm: 5.6.0

yeoman: 1.8.5

yarn: 1.7.0

Docker version 18.03.1-ce, build 9ee9f40

docker-compose version 1.21.1, build 7641a569
`

##### **Browsers and Operating System**
Chrome

- [*] Checking this box is mandatory (this is just to show you read everything)

**Comment by chrisdns** [comment]
It must be related to the fact that when you click the save button on the edit page, you need to scroll down and this is triggering the function that fetches the next page that is causing the bug

That was not expected :sweat_smile:  
I'll try to fix that, thank you for reporting the issue

**Comment by deepu105** [comment]
closed via #7934

**Comment by deepu105** [comment]
@chrisdns plz add the fix #no keyword for autoclose next time
